app/utils/authorization.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import unicode_literals
+
+from functools import wraps
+from flask import request
+
+from app.utils.auth import get_user_from_jwt, handle_error
+from app.mod_api.models import User, MetaDataDB, Publisher, PublisherUser
+
+
+roles = {
+    "Package": {
+        "Owner": ["Package::Read", "Package::Create", "Package::Delete",
+                  "Package::Undelete", "Package::Purge", "Package::Update",
+                  "Package::Tag"],
+        "Editor": ["Package::Read", "Package::Create", "Package::Delete",
+                   "Package::Undelete", "Package::Update", "Package::Tag"],
+        "Viewer": ["Package::Read"]
+    },
+    "Publisher": {
+        "Owner": ["Publisher::AddMember", "Publisher::RemoveMember",
+                  "Publisher::Create", "Publisher::Read", "Publisher::Delete",
+                  "Publisher::Update", "Publisher::ViewMemberList"],
+        "Editor": ["Publisher::ViewMemberList", "Publisher::AddMember",
+                   "Publisher::RemoveMember", "Publisher::Read"],
+        "Viewer": ["Publisher::Read"]
+    },
+    "System": {
+        "LoggedIn": ["Package::Create", "Publisher::Create"],
+        "Anonymous": ["Package::Read"],
+        "Sysadmin": ["Package::Read", "Package::Create", "Package::Delete",
+                     "Package::Undelete", "Package::Purge", "Package::Update",
+                     "Package::Tag", "Publisher::AddMember", "Publisher::RemoveMember",
+                     "Publisher::Create", "Publisher::Read", "Publisher::Delete",
+                     "Publisher::Update", "Publisher::ViewMemberList"]
+    }
+}
+
+
+def is_allowed(action):
+    def wrapper(f):
+        @wraps(f)
+        def wrapped(*args, **kwargs):
+            entity_str, action_str = action.split("::")
+            user_id, instance = None, None
+            jwt_status, user_info = get_user_from_jwt(request)
+            if jwt_status:
+                user_id = user_info['user']
+
+            if entity_str == 'Package':
+                publisher_name, package_name = kwargs['publisher'], kwargs['package']
+                instance = MetaDataDB.get_package(publisher_name, package_name)
+
+            elif entity_str == 'Publisher':
+                publisher_name = kwargs['publisher']
+                instance = Publisher.query.filter_by(name=publisher_name).one()
+            else:
+                return handle_error("INVALID_ENTITY", "{e} is not a valid one".format(e=entity_str), 401)
+            if instance is None:
+                return handle_error("INVALID_INPUT", "Could not find publisher or package", 404)
+            status = is_role_allowed(user_id, instance, action)
+            if not status:
+                return handle_error("NOT_ALLOWED", "The operation is not allowed", 403)
+            return f(*args, **kwargs)
+        return wrapped
+    return wrapper
+
+
+def is_role_allowed(user_id, entity, action):
+    local_roles = get_roles(user_id, entity)
```

Comment:
You can see the example here: https://github.com/btbroot/exmo2010/blob/master/exmo/auth/helpers.py#L56
So it is up to a single routine to perform a check and return true\false.

 It may also be helpful to define methods on the user model to check if the user is owner or member of a publisher, for example: 
```
def is_publisher_owner(self, publisher):
    user_role = PublisherUser.query.join(User).join(Publisher)\
            .filter(User.id == self.id, Publisher.name == publisher.name).one()
    return True if user_role.role == 'OWNER' else False

def is_publisher_member(self, publisher):
     ...
```